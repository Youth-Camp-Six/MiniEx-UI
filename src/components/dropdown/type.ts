import { ReactNode } from 'react';
export type DropdownOptions = {
  label: number | string;
  value: number | string;
};
type align = 'start' | 'end' | 'left' | 'right' | 'center' | 'justify' | 'match-parent';
export interface DropdownProps {
  children: ReactNode;
  className?: string;
  options: DropdownOptions[];
  trigger?: string;
  arrow?: boolean;
  width?: string;
  maxHeight?: string;
  align?: align | undefined;
  placement?: string;
  handleClick?: (e: any) => void;
}
