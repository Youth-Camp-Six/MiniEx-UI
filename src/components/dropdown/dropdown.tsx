import React, { memo } from 'react';
import { DropdownProps, DropdownOptions } from './type';
import { Popover } from '../popover';

export const Dropdown: React.FC<DropdownProps> = (props) => {
  const {
    className,
    children,
    options,
    trigger,
    arrow,
    width,
    maxHeight,
    align,
    placement,
    ...restProps
  } = props;

  const DropdownBody = (arr: DropdownOptions[]) => {
    return (
      <div className='mi-dropdown-body' style={{ width: `${width}`, maxHeight: `${maxHeight}` }}>
        {arr.map((item, index) => {
          return (
            <div key={index} className='mi-dropdown-option' style={{ textAlign: align }}>
              {item.label}
            </div>
          );
        })}
      </div>
    );
  };

  return (
    <Popover
      className={className}
      trigger={trigger}
      arrow={arrow}
      placement={placement}
      {...restProps}
      childrenData={DropdownBody(options)}
    >
      {children}
    </Popover>
  );
};

Dropdown.defaultProps = {
  options: [],
  trigger: 'hover',
  arrow: true,
  width: '',
  maxHeight: '300px',
  align: 'left',
  placement: 'bottom',
};

export default memo(Dropdown);
