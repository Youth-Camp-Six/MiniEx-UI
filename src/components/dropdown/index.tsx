import Dropdown from './dropdown'

export * from './type'
export { Dropdown }