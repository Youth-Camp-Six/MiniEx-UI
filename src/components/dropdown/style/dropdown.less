.mi-dropdown-body {
    display: flex;
    flex-direction: column;
    user-select: none;
    overflow: auto;
    box-sizing: border-box;
    .mi-dropdown-option {
        padding: 0px 10px;
        height: 32px;
        line-height: 32px;
        font-size: 14px;
        border-radius: var(--mi-border-radius);
        color: var(--mi-text-color-7);
        cursor: pointer;
        white-space: nowrap;
        box-sizing: border-box;
    }
    .mi-dropdown-option:hover {
        color: var(--mi-text-color-2);
        background-color: var(--mi-bgcolor-2);
    }
}
.mi-dropdown-body::-webkit-scrollbar {
    background-color: rgb(126, 126, 126, 0);
    width: 5px;
    height: 5px;
}

.mi-dropdown-body::-webkit-scrollbar-thumb:hover {
    background-color: rgb(126, 126, 126);
}
.mi-dropdown-body::-webkit-scrollbar-thumb {
    background-color: rgb(209 213 219 / 0);
    border-radius: 5px;
}
.mi-dropdown-body:hover::-webkit-scrollbar-thumb {
    background-color: rgb(209 213 219 / 1);
    border-radius: 5px;
}