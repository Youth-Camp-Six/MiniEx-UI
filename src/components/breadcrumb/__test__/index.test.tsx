/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Breadcrumb } from '..';
import { Dropdown } from '../../dropdown';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Breadcrumb', () => {
  // 快照测试
  it('test Breadcrumb ', () => {
    const options = [
      {
        label: '首页',
        to: '/',
      },
      {
        label: 'Vue',
        to: 'https://vuejs.org',
      },
      {
        label: '商品列表',
        to: '',
        active: true,
      },
      {
        label: (
          <Dropdown
            options={[
              {
                label: '西游记',
                value: '1',
              },
              {
                label: '水浒传',
                value: '2',
              },
            ]}
          >
            <div>纯文本</div>
          </Dropdown>
        ),
        to: '',
      },
    ];
    const wrapper = render(
      <Router>
        <Breadcrumb options={options}></Breadcrumb>
        <Breadcrumb options={options} iconType='shoulder'></Breadcrumb>
      </Router>
    );
    expect(wrapper.container.firstChild).toMatchSnapshot();
  });
});
