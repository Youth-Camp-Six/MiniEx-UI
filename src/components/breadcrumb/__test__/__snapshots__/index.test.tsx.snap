// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb test Breadcrumb  1`] = `
<div
  class="mi-breadcrumb"
>
  <div
    class="mi-breadcrumb-item "
  >
    <span
      class="mi-breadcrumb-isPath"
    >
      首页
    </span>
    <div
      class="mi-breadcrumb-parting"
    >
      <svg
        fill="none"
        stroke="currentColor"
        stroke-linecap="butt"
        stroke-linejoin="miter"
        stroke-width="4"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M29.506 6.502 18.493 41.498"
        />
      </svg>
    </div>
  </div>
  <div
    class="mi-breadcrumb-item "
  >
    <span
      class="mi-breadcrumb-isPath"
    >
      Vue
    </span>
    <div
      class="mi-breadcrumb-parting"
    >
      <svg
        fill="none"
        stroke="currentColor"
        stroke-linecap="butt"
        stroke-linejoin="miter"
        stroke-width="4"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M29.506 6.502 18.493 41.498"
        />
      </svg>
    </div>
  </div>
  <div
    class="mi-breadcrumb-item mi-breadcrumb-active"
  >
    <span
      class=""
    >
      商品列表
    </span>
    <div
      class="mi-breadcrumb-parting"
    >
      <svg
        fill="none"
        stroke="currentColor"
        stroke-linecap="butt"
        stroke-linejoin="miter"
        stroke-width="4"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M29.506 6.502 18.493 41.498"
        />
      </svg>
    </div>
  </div>
  <div
    class="mi-breadcrumb-item "
  >
    <span
      class=""
    >
      <div
        aria-expanded="false"
        class="mi-popover"
        trigger="hover"
      >
        <label>
          <div
            class="trigger"
          >
            <div>
              纯文本
            </div>
          </div>
        </label>
      </div>
    </span>
  </div>
</div>
`;
